pinot-core/src/main/java/com/linkedin/pinot/core/query/aggregation/function/RangeAggregationFunction.java

```diff
@@ -0,0 +1,196 @@
+
+  }
+
+  @Override
+  public RangePair aggregate(Block docIdSetBlock, Block[] block) {
+    double min = Double.POSITIVE_INFINITY;
+    double max = Double.NEGATIVE_INFINITY;
+    int docId = 0;
+    Dictionary dictionaryReader = block[0].getMetadata().getDictionary();
+    BlockDocIdIterator docIdIterator = docIdSetBlock.getBlockDocIdSet().iterator();
+    BlockSingleValIterator blockValIterator = (BlockSingleValIterator) block[0].getBlockValueSet().iterator();
+
+    while ((docId = docIdIterator.next()) != Constants.EOF) {
+      if (blockValIterator.skipTo(docId)) {
+        int dictionaryIndex = blockValIterator.nextIntVal();
+        if (dictionaryIndex != Dictionary.NULL_VALUE_INDEX) {
+          min = Math.min(min, dictionaryReader.getDoubleValue(dictionaryIndex));
+          max = Math.max(max, dictionaryReader.getDoubleValue(dictionaryIndex));
+        }
+      }
+    }
+    return new RangePair(min, max);
+  }
+
+  @Override
+  public RangePair aggregate(RangePair mergedResult, int docId, Block[] block) {
+    BlockSingleValIterator blockValIterator = (BlockSingleValIterator) block[0].getBlockValueSet().iterator();
+    if (blockValIterator.skipTo(docId)) {
+      int dictId = blockValIterator.nextIntVal();
+      if (dictId != Dictionary.NULL_VALUE_INDEX) {
+        double currentValue = block[0].getMetadata().getDictionary().getDoubleValue(dictId);
+        if (mergedResult == null) {
+          return new RangePair(currentValue, currentValue);
+        }
+        return new RangePair(
+            Math.min(mergedResult.getFirst(), currentValue),
+            Math.max(mergedResult.getSecond(), currentValue));
+      }
+    }
+    return mergedResult;
+  }
+
+  @Override
+  public List<RangePair> combine(List<RangePair> aggregationResultList, CombineLevel combineLevel) {
+    double combinedMinResult = Double.POSITIVE_INFINITY;
+    double combinedMaxResult = Double.NEGATIVE_INFINITY;
+    for (RangePair aggregationResult : aggregationResultList) {
+      combinedMinResult = Math.min(combinedMinResult, aggregationResult.getFirst());
+      combinedMaxResult = Math.max(combinedMaxResult, aggregationResult.getSecond());
+    }
+    aggregationResultList.clear();
+    aggregationResultList.add(new RangePair(combinedMinResult, combinedMaxResult));
+    return aggregationResultList;
+  }
+
+  @Override
+  public RangePair combineTwoValues(RangePair aggregationResult0, RangePair aggregationResult1) {
+    if (aggregationResult0 == null) {
+      return aggregationResult1;
+    }
+    if (aggregationResult1 == null) {
+      return aggregationResult0;
+    }
+    return new RangePair(
+        Math.min(aggregationResult0.getFirst(), aggregationResult1.getFirst()),
+        Math.max(aggregationResult0.getSecond(), aggregationResult1.getSecond()));
+  }
+
+  @Override
+  public Double reduce(List<RangePair> combinedResult) {
+    if (combinedResult.isEmpty()) {
+      return Double.NaN;
```

Comment:
Actually, -1 would be a better value (as max - min is always >= 0). -1 would be a more manageable value.